humphrey/src/tokio/app.rs

```diff
@@ -129,51 +133,65 @@ where
         let error_handler = Arc::new(self.error_handler);
 
         loop {
-            match socket.accept().await {
-                Ok((mut stream, _)) => {
-                    let cloned_state = self.state.clone();
-
-                    // Check that the client is allowed to connect
-                    if (self.connection_condition)(&mut stream, cloned_state) {
-                        let cloned_state = self.state.clone();
-                        let cloned_monitor = self.monitor.clone();
-                        let cloned_subapps = subapps.clone();
-                        let cloned_default_subapp = default_subapp.clone();
-                        let cloned_error_handler = error_handler.clone();
-
-                        cloned_monitor.send(
-                            Event::new(EventType::ConnectionSuccess)
-                                .with_peer_result(stream.peer_addr()),
-                        );
-
-                        // Spawn a new thread to handle the connection
-                        tokio::spawn(async move {
-                            cloned_monitor.send(
-                                Event::new(EventType::ThreadPoolProcessStarted)
-                                    .with_peer_result(stream.peer_addr()),
-                            );
+            let shutdown = async {
+                if let Some(sd) = self.shutdown.clone() {
+                    sd.cancelled().await
+                } else {
+                    loop {
+                        let _ = tokio::time::sleep(std::time::Duration::from_secs(9999));
```

Comment:
If you're trying to express "future that never resolves" you should use [`future::pending`](https://docs.rs/futures/latest/futures/future/fn.pending.html) I think.